File: output/GPT3.5-Turbo/xCodeEval_4000/C++/C/a807aea37e97bb68cb5d69671f1b9f72.c
#include <stdio.h>
#include <stdlib.h>

#define MX 5000

void solve() {
    int N;
    scanf("%d", &N);
    int K[N];
    for (int i = 0; i < N; i++) {
        scanf("%d", &K[i]);
    }
    // Sort the array
    for (int i = 0; i < N; i++) {
        for (int j = i + 1; j < N; j++) {
            if (K[i] > K[j]) {
                int temp = K[i];
                K[i] = K[j];
                K[j] = temp;
            }
        }
    }
    int C[MX+1] = {0};
    for (int i = 0; i < N; i++) {
        C[K[i]]++;
    }

    int P[MX+1] = {0};
    int primes[15] = {2,3,5,7,11,13,17,19,23,29,31,37,41,43,47};
    int M = 15;

    int D[MX+1][15] = {0};
    for (int i = 0; i < M; i++) {
        int p = primes[i];
        for (int p = primes[i]; p <= MX; p *= primes[i]) {
            for (int j = 0; j <= MX; j++) {
                D[j][i] += j / p;
            }
        }
    }
    for (int i = 0; i <= MX; i++) {
        for (int j = 0; j < M; j++) {
            int temp = D[i][j];
            D[i][j] = D[i][M-j-1];
            D[i][M-j-1] = temp;
        }
    }

    long long Pref[MX+1][16] = {0};
    for (int i = 0; i <= MX; i++) {
        for (int j = 0; j < M; j++) {
            Pref[i][j+1] = Pref[i][j] + D[i][j];
        }
    }

    int CountPref[MX+2] = {0};
    long long tot = 0;
    for (int i = 0; i <= MX; i++) {
        CountPref[i+1] = CountPref[i] + C[i];
        tot += C[i] * Pref[i][M];
    }

    int use = K[N/2];
    long long W[N];
    long long base = 0;
    int index = 0;
    for (int i = 0; i <= MX; i++) {
        if (C[i] == 0) continue;

        int x = 0;
        long long len = 0;
        while (x < M && D[i][x] == D[use][x]) {
            len += D[i][x];
            x++;
        }
        if (x < M) {
            len += D[i][x] < D[use][x] ? D[i][x] : D[use][x];
        }
        W[index++] = len;
        base += (Pref[i][M] - len) * C[i];
    }

    long long ans = 1e18;
    for (int i = 0; i < index; i++) {
        long long cur = 0;
        for (int j = 0; j < index; j++) {
            cur += abs(W[j] - W[i]) * C[j];
        }
        if (cur < ans) {
            ans = cur;
        }
    }
    printf("%lld\n", ans);
}

int main() {
    solve();
    return 0;
}

File: output/GPT3.5-Turbo/xCodeEval_4000/C++/Java/a807aea37e97bb68cb5d69671f1b9f72.java
import java.util.*;
import java.io.*;

class DChaoticV {
    public void solve(Scanner cin, PrintWriter cout) {
        int N = cin.nextInt();
        List<Integer> K = new ArrayList<>();
        for (int i = 0; i < N; i++) {
            K.add(cin.nextInt());
        }
        Collections.sort(K);
        List<Integer> C = new ArrayList<>(Collections.nCopies(5001, 0));
        for (int k : K) {
            C.set(k, C.get(k) + 1);
        }

        List<Integer> P = new Sieve(5000).primes();
        int M = P.size();

        List<List<Integer>> D = new ArrayList<>();
        for (int i = 0; i <= 5000; i++) {
            D.add(new ArrayList<>(Collections.nCopies(M, 0)));
        }
        for (int i = 0; i < M; i++) {
            int p = P.get(i);
            for (int j = p; j <= 5000; j += p) {
                for (int k = 0; k <= 5000; k++) {
                    D.get(k).set(i, D.get(k).get(i) + k / p);
                }
            }
        }
        for (int i = 0; i <= 5000; i++) {
            Collections.reverse(D.get(i));
        }

        List<List<Long>> Pref = new ArrayList<>();
        for (int i = 0; i <= 5000; i++) {
            Pref.add(new ArrayList<>(Collections.nCopies(M + 1, 0L)));
        }
        for (int i = 0; i <= 5000; i++) {
            for (int j = 0; j < M; j++) {
                Pref.get(i).set(j + 1, Pref.get(i).get(j) + D.get(i).get(j));
            }
        }

        List<Integer> CountPref = new ArrayList<>(Collections.nCopies(5002, 0));
        long tot = 0;
        for (int i = 0; i <= 5000; i++) {
            CountPref.set(i + 1, CountPref.get(i) + C.get(i));
            tot += C.get(i) * Pref.get(i).get(M);
        }

        int use = K.get(N / 2);
        List<Pair<Long, Integer>> W = new ArrayList<>();
        long base = 0;
        for (int i = 0; i <= 5000; i++) {
            if (C.get(i) == 0) continue;

            int x = 0;
            long len = 0;
            while (x < M && D.get(i).get(x) == D.get(use).get(x)) {
                len += D.get(i).get(x);
                x++;
            }
            if (x < M) {
                len += Math.min(D.get(i).get(x), D.get(use).get(x));
            }
            W.add(new Pair<>(len, C.get(i)));
            base += (Pref.get(i).get(M) - len) * C.get(i);
        }
        Collections.sort(W, new Comparator<Pair<Long, Integer>>() {
            public int compare(Pair<Long, Integer> a, Pair<Long, Integer> b) {
                return a.x.compareTo(b.x);
            }
        });

        long ans = (long) 1e18;
        for (Pair<Long, Integer> w : W) {
            long cur = 0;
            for (Pair<Long, Integer> ww : W) {
                cur += Math.abs(ww.x - w.x) * ww.y;
            }
            ans = Math.min(ans, base + cur);
        }
        cout.println(ans);
    }

    public static void main(String[] args) throws Exception {
        Scanner cin = new Scanner(System.in);
        PrintWriter cout = new PrintWriter(System.out);
        DChaoticV solver = new DChaoticV();
        solver.solve(cin, cout);
        cout.close();
    }

    static class Pair<T, U> {
        T x;
        U y;

        Pair(T x, U y) {
            this.x = x;
            this.y = y;
        }
    }

    static class Sieve {
        boolean[] prime;

        Sieve(int n) {
            prime = new boolean[n + 1];
            Arrays.fill(prime, true);
            prime[0] = prime[1] = false;
            for (int i = 2; i * i <= n; i++) {
                if (prime[i]) {
                    for (int j = i * i; j <= n; j += i) {
                        prime[j] = false;
                    }
                }
            }
        }

        List<Integer> primes() {
            List<Integer> ans = new ArrayList<>();
            for (int i = 2; i < prime.length; i++) {
                if (prime[i]) {
                    ans.add(i);
                }
            }
            return ans;
        }
    }
}
